### PR TITLE
AIP-64: Add task instance history list endpoint

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This endpoint lists the task instance histories. It is more like the task instance list endpoint but for the history. This will be used to replace the UI task instance history endpoint

### CODE REVIEW

File: airflow/www/static/js/types/api-generated.ts [hunk 1]
```diff
@@ -4347,6 +4397,69 @@ export interface operations {
...
     };
   };
+  /** This endpoint allows specifying `~` as the dag_id, dag_run_id to retrieve DAG runs for all DAGs and DAG runs. */
+  get_task_instances_history: {
+    parameters: {
+      path: {
+        /** The DAG ID. */
+        dag_id: components["parameters"]["DAGID"];
+        /** The DAG run ID. */
+        dag_run_id: components["parameters"]["DAGRunID"];
```
Comment (bbovenzi) [inline airflow/www/static/js/types/api-generated.ts:4391]: Where do I specify the task_id and map_index?

Comment (ephraimbuddy) [reply airflow/www/static/js/types/api-generated.ts:4391]: If that's what you want, then the previous endpoint would do. See: https://github.com/apache/airflow/pull/40441/files#diff-d4d4076dfd54f6dd28663f4677a8a5037cfddf9366784d4e7a46a39e08c4ec48R4290-R4299.

This endpoint gets all the task instance history in a dag_run, all the TI history in a dag or all the TI history in airflow. 

Let me know if the previous one is what we need and not this

Comment (bbovenzi) [reply airflow/www/static/js/types/api-generated.ts:4391]: But what if I want a list of all tries for a single task instance?

Comment (ephraimbuddy) [reply airflow/www/static/js/types/api-generated.ts:4391]: Oh, got it. I will modify this to fit

Comment (ephraimbuddy) [reply airflow/www/static/js/types/api-generated.ts:4391]: Updated it

Comment (bbovenzi) [reply airflow/www/static/js/types/api-generated.ts:4391]: We're still missing `map_index`

Comment (ephraimbuddy) [reply airflow/www/static/js/types/api-generated.ts:4391]: There're two endpoints. One includes the map_index.

Comment (ephraimbuddy) [reply airflow/www/static/js/types/api-generated.ts:4391]: Check below

**Review by jedcunningham — APPROVED** [review]

**Review by bbovenzi — APPROVED** [review]

**Pull request merged**